File: frontend/src/contracts/documentation.ts
export interface DocumentationPageSummary {
  id: string;
  title: string;
  description: string;
}

export interface DocumentationSection {
  id: string;
  title: string;
  pages: DocumentationPageSummary[];
}

export interface DocumentationIndexResponse {
  schema_version: '1.0';
  default_page_id: string | null;
  sections: DocumentationSection[];
}

export interface DocumentationPageResponse {
  schema_version: '1.0';
  id: string;
  title: string;
  section_id: string;
  section_title: string;
  content: string;
}

File: backend/app/contracts/documentation.py
from __future__ import annotations

from typing import Literal, cast

from pydantic import BaseModel, Field


class DocumentationPageSummary(BaseModel):
    schema_version: Literal["1.0"] = "1.0"
    id: str
    title: str
    section_id: str
    order: int = 0


class DocumentationSection(BaseModel):
    schema_version: Literal["1.0"] = "1.0"
    id: str
    title: str
    order: int = 0
    pages: list[DocumentationPageSummary] = Field(
        default_factory=lambda: cast(list["DocumentationPageSummary"], [])
    )


class DocumentationNavigationResponse(BaseModel):
    schema_version: Literal["1.0"] = "1.0"
    documentation_version: str
    # field expected by frontend: `default_page_id`
    default_page_id: str
    sections: list[DocumentationSection] = Field(
        default_factory=lambda: cast(list["DocumentationSection"], [])
    )


class DocumentationPageResponse(BaseModel):
    schema_version: Literal["1.0"] = "1.0"
    id: str
    title: str
    section_id: str
    # frontend expects `content` key containing markdown
    content: str
    source_path: str
    documentation_version: str
